lua/dylib.lua

```diff
@@ -30,6 +71,81 @@ int add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
 	enum gk_fib_action action, struct gk_config *gk_conf);
 int del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf);
 
+struct ipaddr *get_fib_dump_entry_ip(struct gk_fib_dump_entry *entry);
+struct ipaddr *get_fib_dump_entry_grantor_ip(struct gk_fib_dump_entry *entry);
+struct ipaddr *get_fib_ether_dump_entry_nexthop(
+	struct gk_fib_ether_dump_entry *entry);
+struct ether_addr *get_fib_ether_dump_entry_mac(
+	struct gk_fib_ether_dump_entry *entry);
+int convert_ip_to_str(struct ipaddr *ip_addr, char *res, int n);
+void ether_format_addr(char *buf, uint16_t size,
+	const struct ether_addr *eth_addr);
+
 ]]
 
 c = ffi.C
+
+function print_fib_dump_entry(fib_dump_entry, arg)
+	local i = 0
+	local ret
+	local max_ip_addr_len = 128
+	local entry = ffi.cast("struct gk_fib_dump_entry *", fib_dump_entry)
+	local ip_addr = ffi.new("char[?]", max_ip_addr_len)
+
+	local grantor
+	local ip = dylib.c.get_fib_dump_entry_ip(entry)
+	ret = dylib.c.convert_ip_to_str(ip, ip_addr, max_ip_addr_len)
+	if ret < 0 then
+		return -1
+	end
+
+	print("FIB entry for IP prefix: " + ip_addr + "/" + str(entry.len) +
+		" with action " + str(entry.action));
+
+	if entry.action ~= dylib.c.GK_FWD_GRANTOR then
+		goto ether_cache_tbl
+	end
```

Comment:
This `goto` is mostly adding complexity. Just invert the test (i.e. `entry.action == dylib.c.GK_FWD_GRANTOR`) and move the code below into the body of the new `if` statement.